multihost/db-postgresql-with-custom-certificates/test.sh

```diff
@@ -0,0 +1,375 @@
+        rlRun "sudo -u postgres psql -f setup.psql"
+        rlRun "sync-set ATTESTATION_SERVER_GENERATE_CERTS_DONE"
+
+        # tenant config
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+        rlRun "limeUpdateConf tenant verifier_ip $ATTESTATION_SERVER_IP"
+        rlRun "limeUpdateConf tenant registrar_ip $ATTESTATION_SERVER_IP"
+        rlRun "limeUpdateConf tenant tls_dir $CERTDIR"
+        rlRun "limeUpdateConf tenant trusted_server_ca '[\"cacert.pem\"]'"
+        rlRun "limeUpdateConf tenant client_cert tenant-cert.pem"
+        rlRun "limeUpdateConf tenant client_key tenant-key.pem"
+
+        #copy cert to agent machine
+        rlRun "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null $CERTDIR/cacert.pem root@$AGENT_IP:/var/lib/keylime"
+        rlRun "sync-block ANSIBLE_SETUP_DONE ${SETUP_SERVER_IP}" 0 "Waiting for ansible setup of system roles."
+        sleep 5
+        rlRun "limeWaitForVerifier"
+        rlRun "limeWaitForRegistrar"
+
+        #confirm that verifier and registrar accessed and modified databases
+        rlRun -s "sudo -u postgres psql -c 'SELECT datname FROM pg_database;'"
+        rlAssertGrep "verifierdb" $rlRun_LOG
+        rlAssertGrep "registrardb" $rlRun_LOG
+
+        rlRun "sync-set ATTESTATION_SERVER_START"
+        rlRun "sync-block AGENT_STARTS ${AGENT_IP}" 0 "Waiting for ansible setup of system roles."
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+    rlPhaseEnd
+
+    rlPhaseStartTest "keylime attestation test: Add Agent"
+        # register AGENT and confirm it has passed validation
+        rlRun "cat > script.expect <<_EOF
+set timeout 20
+spawn keylime_tenant -v ${ATTESTATION_SERVER_IP} -t ${AGENT_IP} -u ${AGENT_ID} --runtime-policy /var/tmp/policy.json --include payload-${REVOCATION_SCRIPT_TYPE} --cert default -c add
+expect \"Please enter the password to decrypt your keystore:\"
+send \"keylime\n\"
+expect eof
+_EOF"
+        rlRun "expect script.expect"
+        rlRun "limeWaitForAgentStatus $AGENT_ID 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'$AGENT_ID'" $rlRun_LOG -E
+        rlRun "sync-set AGENT_ADD_DONE"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Agent attestation test: Fail keylime agent"
+        rlRun "limeWaitForAgentStatus $AGENT_ID '(Failed|Invalid Quote)'"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Verifier test"
+        # check that the AGENT failed verification
+        rlAssertGrep "WARNING - File not found in allowlist: .*/keylime-bad-script.sh" $(limeVerifierLogfile) -E
+        rlRun "cat $(limeVerifierLogfile)"
+        AGENT_ID="d432fbb3-d2f1-4a97-9ef7-75bd81c00000"
+        rlAssertGrep "WARNING - Agent $AGENT_ID failed, stopping polling" $(limeVerifierLogfile)
+    rlPhaseEnd
+
+    rlPhaseStartCleanup "Attestation server cleanup"
+        rlRun "sync-block AGENT_ALL_TESTS_DONE ${AGENT_IP}" 0 "Waiting for the Agent to finish the test"
```

Comment:
I think that  once we have semaphores above `AGENT_ALL_TESTS_DONE` is not needed anymore.

Reply:
Above checking if agent failed on verifier side?

Reply:
I mean, once we have AGENT_CORRUPTED and AGENT_FAILED then AGENT_ALL_TESTS_DONE is not needed because it is not blocking anything important. Is it? It seems to block only the cleanup but that can happen independently on each system since the agent failure has been already processed by the verifier (due to AGENT_FAILED set).